libs/core/src/main/java/com/sourcefuse/jarc/core/filters/constants/OperationPredicates.java

```diff
@@ -0,0 +1,129 @@
+package com.sourcefuse.jarc.core.filters.constants;
+
+import static java.util.Map.entry;
+
+import com.sourcefuse.jarc.core.filters.interfaces.TriFunction;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.Expression;
+import jakarta.persistence.criteria.Predicate;
+import java.util.List;
+import java.util.Map;
+
+public final class OperationPredicates {
+
+  private static Map<String, TriFunction<CriteriaBuilder, 
+  Expression<String>, Object, Predicate>> operatorMap =
+    Map.ofEntries(
+      entry("eq", CriteriaBuilder::equal),
+      entry("neq", CriteriaBuilder::notEqual),
+      entry(
+        "in",
+        (
+            CriteriaBuilder criteriaBuilder,
+            Expression<String> path,
+            Object value
+          ) ->
+          path.in((List<?>) value)
+      ),
+      entry(
+        "nin",
+        (
+            CriteriaBuilder criteriaBuilder,
+            Expression<String> path,
+            Object value
+          ) ->
+          criteriaBuilder.not(path.in((List<?>) value))
+      ),
+      entry(
+        "like",
+        (
+            CriteriaBuilder criteriaBuilder,
+            Expression<String> path,
+            Object value
+          ) ->
+          criteriaBuilder.like(path, "%" + value + "%")
```

Comment:
the value should have the % why do we have to add this, same goes with not like

Reply:
If we want user to pass the value with %, then we can remove the % from code